src/steps/index.test.ts

```diff
@@ -133,3 +137,72 @@ test('should collect data', async () => {
     },
   });
 });
+
+describe('stopFetchingRequests', () => {
```

Comment:
Thanks for taking up the work to break the tests out. 👍   I'm going to go ahead and merge these changes, but I thought you might like some additional food for thought.

Ideally, each test could be run concurrently and share no state whatsoever. Generally, this means that there would be no variables shared across the `test()` functions, as the `requests`, `lastExecutionTime`, and `templateRequest` are being shared.

1. `lastExecutionTime` - no need for a variable at all, just pass in the value. The variable name is helpful for understanding what the second argument is for the `stopFetchingRequests`, but in the context of a file devoted to testing the function, it is pretty easy to get that from the context.
2. `requests` - just pass in such as `expect(stopFetchingRequests([], 0)).toBe(true);`, or `expect(stopFetchingRequests([buildRequest({ updatedAt: getDaysAgoDateString(13) })], 0)).toBe(true);`. This would drive the development of another test: `test('last request in response is considered oldest')`, where there are two requests with the 0th item have a different timestamp than expected, to be sure it isn't used.
3. `templateRequest` - use a `buildRequest()` function to generate a new value based on defaults (those in the current `templateRequest`). This will avoid mutating a single instance which would be a race condition when executing concurrently.